apps/desktop/src/lib/config/appSettings.ts

```diff
@@ -2,99 +2,87 @@
-	return persisted(false, 'appAnalyticsConfirmed');
-}
-
-/**
- * Provides a writable store for obtaining or setting the current state of application metrics.
- * The application metrics can be enabled or disabled by setting the value of the store to true or false.
- * @returns A writable store with the appMetricsEnabled config.
- */
-export function appMetricsEnabled() {
-	return persisted(true, 'appMetricsEnabled');
-}
-
-/**
- * Provides a writable store for obtaining or setting the current state of application error reporting.
- * The application error reporting can be enabled or disabled by setting the value of the store to true or false.
- * @returns A writable store with the appErrorReportingEnabled config.
- */
-export function appErrorReportingEnabled() {
-	return persisted(true, 'appErrorReportingEnabled');
-}
-
-/**
- * Provides a writable store for obtaining or setting the current state of non-anonemous application metrics.
- * The setting can be enabled or disabled by setting the value of the store to true or false.
- * @returns A writable store with the appNonAnonMetricsEnabled config.
- */
-export function appNonAnonMetricsEnabled() {
-	return persisted(false, 'appNonAnonMetricsEnabled');
-}
-
-function persisted<T>(initial: T, key: string): Writable<T> & { onDisk: () => Promise<T> } {
-	async function setAndPersist(value: T, set: (value: T) => void) {
-		await store.set(key, value);
-		await store.save();
-
-		set(value);
-	}
-
-	async function synchronize(set: (value: T) => void): Promise<void> {
-		const value = await storeValueWithDefault(initial, key);
-		set(value);
-	}
-
-	function update() {
-		throw 'Not implemented';
+export class AppSettings {
+	diskStore: Store | undefined;
+	constructor() {
+		Store.load('settings.json', { autoSave: true })
+			.then((store) => {
+				this.diskStore = store;
+			})
+			.catch((e) => {
+				console.error(e);
+			});
 	}
 
-	const thisStore = writable<T>(initial, (set) => {
-		synchronize(set);
-	});
-
-	async function set(value: T) {
-		setAndPersist(value, thisStore.set);
-	}
-
-	async function onDisk() {
-		return await storeValueWithDefault(initial, key);
-	}
-
-	const subscribe = thisStore.subscribe;
-
-	return {
-		subscribe,
-		set,
-		update,
-		onDisk
-	};
-}
-
-async function storeValueWithDefault<T>(initial: T, key: string): Promise<T> {
-	try {
-		await store.load();
-	} catch {
-		// If file does not exist, reset it
-		store.reset();
+	/**
+	 * Persisted confirmation that user has confirmed their analytics settings.
+	 */
+	readonly appAnalyticsConfirmed = this.persisted(false, 'appAnalyticsConfirmed');
+
+	/**
+	 * Provides a writable store for obtaining or setting the current state of application metrics.
+	 * The application metrics can be enabled or disabled by setting the value of the store to true or false.
+	 * @returns A writable store with the appMetricsEnabled config.
+	 */
+	readonly appMetricsEnabled = this.persisted(true, 'appMetricsEnabled');
```

Comment:
these still work correctly, right?

Reply:
yes, but let me also test this manually before landing these changes.

Reply:
I need to investigate this before we merge.